File: Number_Guessing_Games.py
import random

# Gera um número aleatorio entre 1 e 100
numero_secreto = random.randint(1, 100)
print('Vamos jogar um jogo de adivinhação, você podera chutar quantas vezes você quiser, você tera que chutar um número entre 1 e 100, vamos ver a sua sorte nisso')

# Inicia o loop do jogo
while True:
    # Obtém a tentativa do usuário
    tentativa = int(input('Tente adivinhar o número (entre 1 e 100): '))

    # Verifica se a tentativa está correta
    if tentativa == numero_secreto:
        print('Parabéns! Você acertou!')
        break  # Sai do loop, pois o jogo terminou
    elif tentativa < numero_secreto:
        print('Tente um número maior.')
    else:
        print('Tente um número menor.')

File: Number_Guessing_Games.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main() {
    srand(time(0));
    int numeroSecreto = rand() % 100 + 1;
    int tentativas = 0;

    while (true) {
        std::cout << "Digite um número: ";
        int tentativa;
        std::cin >> tentativa;
        tentativas++;

        if (tentativa == numeroSecreto) {
            std::cout << "Parabéns! Você acertou o número em " << tentativas << " tentativas." << std::endl;
            break;
        } else if (tentativa < numeroSecreto) {
            std::cout << "Tente um número maior." << std::endl;
        } else {
            std::cout << "Tente um número menor." << std::endl;
        }
    }

    return 0;
}
